### PR TITLE
Refactor create_worker, minor code tweaks

### PR DESCRIPTION
**Pull request by jajik**
Addresses #102 . There are still functions worthy of attention (hence the Draft marking) such as `internal_find_best_byrequests` or `update_workers_lbstatus`.

### CODE REVIEW

**Comment by jajik** [comment]
Marking ready to review.

I tried to split three very big functions (`create_worker`, `internal_find_best_byrequests`, `update_workers_lbstatus`) into smaller ones and although there is still room for improvements, I think we can go with this. I also removed some unnecessary else branches after return.

**Comment by jajik** [comment]
(I've just rebased to the latest main.)

File: native/mod_proxy_cluster/mod_proxy_cluster.c [hunk 1]
```diff
@@ -1325,6 +1428,7 @@ static void update_workers_lbstatus(proxy_server_conf *conf, apr_pool_t *pool, s
     /* update lbstatus if needed */
     for (i = 0; i < size; i++) {
         nodeinfo_t *ou;
+        int cont;
```
Comment (jfclere) [inline native/mod_proxy_cluster/mod_proxy_cluster.c:1431]: Why here? and not in the if (child_stopping) { }?

**Review by jfclere — APPROVED** [review]

File: native/mod_proxy_cluster/mod_proxy_cluster.c (hunk 1 above)
Comment (jfclere) [reply native/mod_proxy_cluster/mod_proxy_cluster.c:1431]: if () {
    int count;
    ....
}
is OK

if () {
    ...
    int count;
   ...
}
is NOT.

Comment (jajik) [reply native/mod_proxy_cluster/mod_proxy_cluster.c]: Well, it turned out I was wrong :) Fixed, thank you @jfclere !

**Pull request merged**